src/components/SignupPage/SignUp.js

```diff
@@ -52,21 +49,22 @@ export function SignUp({ setSessionObject } ) {
         data.confirmPass = formProps.signUpConfirmPassword;
 
         if (data.password.length < minNumberofChars || data.password.length > maxNumberofChars) {
-            setErrorMessage('The length of the Password should be between 8 and 16 characters');
+            setErrorMessage('');
             setState(false);
+            setErrorMessage('Password length should be between 8 and 16 characters');
+            return <p display='block' className='error-message'>{errorMessage}</p> 
```

Comment:
This can just be an empty return (returning nothing) we just want to prevent the execution of signUpAdapter if there's an error. Please fix for the other cases as well.

Reply:
left return empty as suggested